hathor/transaction/util.py

```diff
@@ -48,13 +51,11 @@ def unpack_len(n: int, buf: bytes) -> tuple[bytes, bytes]:
     return buf[:n], buf[n:]
 
 
-def get_deposit_amount(mint_amount: int) -> int:
-    settings = get_global_settings()
+def get_deposit_amount(settings: HathorSettings, mint_amount: int) -> int:
     return ceil(abs(settings.TOKEN_DEPOSIT_PERCENTAGE * mint_amount))
 
 
-def get_withdraw_amount(melt_amount: int) -> int:
-    settings = get_global_settings()
+def get_withdraw_amount(settings: HathorSettings, melt_amount: int) -> int:
```

Comment:
Should this method receive the percentage and let the caller pass it?

Reply:
Same as https://github.com/HathorNetwork/hathor-core/pull/1085#discussion_r1679874120

Reply:
Agreed. Thanks!